src/common/api/main/PageContextLoginListener.ts

```diff
@@ -36,10 +36,16 @@ export class PageContextLoginListener implements LoginListener {
 		// Update the credentials after the full login.
 		// It is needed because we added encryptedPassphraseKey to credentials which is only
 		// available after the full login which happens async.
-		const storedCredentials = await this.credentialsProvider.getDecryptedCredentialsByUserId(credentials.userId)
-		if (storedCredentials != null) {
-			const updatedCredentials = credentialsToUnencrypted(credentials, storedCredentials.databaseKey)
-			await this.credentialsProvider.store(updatedCredentials)
+
+		// First try to fetch credentials info and only then try to get decrypted credentials as it is not valid to
+		// call `getDecryptedCredentialsByUserId()` without storing any credentials first.
```

Comment:
Just a question: can we just check the key exists? Because I think the key is not tied directly to `CredentialsInfo`?

Reply:
we can either get all of them and search or we can get one of them, that's kind of our options?